src/snowflake/connector/proxy.py

```diff
@@ -39,4 +40,7 @@ def set_proxies(proxy_host, proxy_port, proxy_user=None, proxy_password=None):
         }
         os.environ['HTTP_PROXY'] = proxies['http']
         os.environ['HTTPS_PROXY'] = proxies['https']
+    warn("Arguments `proxy_host`, `proxy_port`, `proxy_user`"
```

Comment:
Maybe like this? Keep the old method but remove from docs and raise a warning?

Reply:
No, sorry. I meant that we instruct customers to possibly set Proxies like this: https://docs.snowflake.com/en/user-guide/python-connector-example.html#using-a-proxy-server

For this change to not be a behavior change we should read out these 2 settings out of the environmental variables and then pass them to `requests`.

Because if someone set up a proxy with env variables months ago then the new connector version should still respect those settings.

Reply:
Ahh, the reason why the tests are failing is because this function is called every time a connection is created and whenever this function is called even with `None`s it will raise a warning.
I'd need an actual approval to introduce a deprecation warning, would it be possible to remove this warning for now and put the information about the `proxies` connection parameter into `SnowflakeConnection`'s docstring?

Reply:
warning removed